buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1095,7 +1084,6 @@ public int writeBytes(InputStream in, int length)
 
     @Override
     public int writeBytes(ScatteringByteChannel in, int length) throws IOException {
-        ensureAccessible();
         ensureWritable(length);
         int writtenBytes = setBytes(writerIndex, in, length);
```

Comment:
the implementation of this method may call `checkIndex` but can now call `checkIndex0`. For example `PooledDirectByteBuf#setBytes(..)` and `PooledHeapByteBuf#setBytes(..)` call `checkIndex`. We should inspect all usages of `checkIndex` in buffer implementations which inherit from `AbstractByteBuf` as suggested in https://github.com/netty/netty/pull/6953#discussion_r126544205.